test/test.cpp

```diff
@@ -3,10 +3,378 @@ bundle config --local path vendor/bundle
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'8', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(8, 2);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'0', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(9, 5);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'1', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(9, 4);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'4', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(9, 3);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'7', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(9, 2);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'*', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+// }
+
+unittest(highlevel) 
+{
+    GodmodeState *state;
+    state = GODMODE();
+    for (int i = 2; i <= 5; ++i)
+    {
+        std::cout << i << " " << (int)(state->digitalPin[i]) << std::endl;
+    }
+
+    const byte ROWS = 4; //four rows
+    const byte COLS = 4; //three columns
+    char keys[ROWS][COLS] = {
+        {'1', '2', '3', 'A'},
+        {'4', '5', '6', 'B'},
+        {'7', '8', '9', 'C'},
+        {'*', '0', '#', 'D'}};
+    byte rowPins[ROWS] = {5, 4, 3, 2}; //connect to the row pinouts of the keypad
+    byte colPins[COLS] = {9, 8, 7, 6}; //connect to the column pinouts of the keypad
+
+    Keypad keypad = Keypad(makeKeymap(keys), rowPins, colPins, ROWS, COLS);
+assertEqual(keypad.countRows(), ROWS);
+assertEqual(keypad.countCols(), COLS);
+
+
+    BitCollector *pin;
+    pin = new BitCollector(6, 5);
```

Comment:
Remind me what the (6, 5) means. Is it what specifies a row/column? If so, then add a comment of "This simulates pressing row 0, column 3, the 'A' key"